src/main/java/co/cask/hydrator/plugin/batch/HTTPSink.java

```
@@ -0,0 +1,402 @@
+      }
+      messages.append(message).append(config.delimiterForMessages);
+    }
+    StringTokenizer tokens = new StringTokenizer(messages.toString().trim(), config.delimiterForMessages);
+    if (config.batchSize == 1 || tokens.countTokens() == config.batchSize) {
+      do {
+        HttpURLConnection conn = null;
+        Map<String, String> headers = config.getRequestHeadersMap();
+        try {
+          URL url = new URL(config.url);
+          conn = (HttpURLConnection) url.openConnection();
+          if (conn instanceof HttpsURLConnection) {
+            //Disable SSLv3
+            System.setProperty("https.protocols", "TLSv1,TLSv1.1,TLSv1.2");
+            if (config.disableSSLValidation) {
+              disableSSLValidation();
+            }
+          }
+          conn.setRequestMethod(config.method.toUpperCase());
+          conn.setConnectTimeout(config.connectTimeout);
+          conn.setReadTimeout(config.readTimeout);
+          conn.setInstanceFollowRedirects(config.followRedirects);
+          conn.addRequestProperty("charset", config.charset);
+          for (Map.Entry<String, String> propertyEntry : headers.entrySet()) {
+            conn.addRequestProperty(propertyEntry.getKey(), propertyEntry.getValue());
+          }
+          if (config.method.equals("POST") || config.method.equals("PUT")) {
+            if (!headers.containsKey("Content-Type")) {
+              conn.addRequestProperty("Content-Type", contentType);
+            }
+          }
+
+          if (messages.length() > 0) {
+            conn.setDoOutput(true);
+            try (OutputStream outputStream = conn.getOutputStream()) {
+              LOG.info("Payload: " + messages.toString().trim());
+              outputStream.write(messages.toString().trim().getBytes(config.charset));
+            }
+          }
+          responseCode = conn.getResponseCode();
+          messages.setLength(0);
+          if (config.failOnNon200Response && responseCode != 200) {
```

Please confirm.
If failOnNon200Response is true and response code is in the range of 201 to 300,we need to throw the exception.